### ISSUE TITLE
New rule: UseStaticImport

### ISSUE THREAD

**Issue by ychulovskyy** [checkstyle/checkstyle#3465]
We found that using static import for Stream API collectors is very readable.

```
...
List<String> list = people.stream()
    .map(Person::getName)
    .collect(toList());
....
```

It's much better than writing:

```
...
List<String> list = people.stream()
    .map(Person::getName)
    .collect(Collectors.toList());
....
```

We would like to have  a rule to notify developer about best practices.

So that:
/var/tmp $ java -jar checkstyle-all.jar -c config.xml .

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">

        <!-- Checks for imports                              -->
        <!-- See http://checkstyle.sf.net/config_import.html -->
        <module name="UseStaticImport">
            <property name="avoid" value="Collectors.toList(),Collectors.toSet()"/>
        </module>
    </module>
</module>
```

Should show violation:
"Use static import for Collectors.toList()"
##

**Comment by MaksimP** [comment]
I think it is more correct to check in PMD.

**Comment by romani** [comment]
Checkstyle is single file valudation tool for now with additiobal limitation for no information about types.
All list of limitations - http://checkstyle.sourceforge.net/writingchecks.html#Limitations

Any implementations of this rule in currend design  will lead to tons of false-positives or over complicated config because it will be required to reguster all static methods.
Matching methods by paramaters is also imposibbe for checkstyle (no types hierarchy information)
